js/ui/codeView.js

```diff
@@ -61,6 +61,10 @@ function _ensureAfterFirstFrame(win, callback) {
 
 function _getAppId(win) {
     let app = Shell.WindowTracker.get_default().get_window_app(win);
+    let gtkId = win.get_gtk_application_id();
```

Comment:
Should this change in `_getAppId()` be split into a separate commit to go upstream? Is it a general fix for flatpaks that ship more than one app?

Reply:
I don't know if this will be interesting for upstream. This does the work for gtk apps that are inside the same flatpak with different desktop files, so the desktop file is not exported. This is a rare case that shouldn't happen in usual apps. We've this problem because we've the `com.endlessm.HackUnlock` and `com.endlessm.HackToolbox` inside the katamari and we won't export the desktop files, but if a flatpak has several apps it usually uses the same prefix and exports all the .desktop files, like libreoffice does.

So I think this is something to work with our especial case flatpak deployment.